inc/class-wp-saml-auth.php

```diff
@@ -129,7 +129,7 @@ public function filter_authenticate( $user, $username, $password ) {
 	 * Do the SAML authentication dance
 	 */
 	public function do_saml_authentication() {
-		$this->provider->requireAuth();
+		$this->provider->requireAuth( array( 'ReturnTo' => $_SERVER['REQUEST_URI'] ) );
```

Comment:
> can either of you think of a downside to directly specifying the ReturnTo URL instead of relying on they deriving in SimpleSAMLphp?

Nope.

> Is there a WordPress abstraction that should be used instead of `$_SERVER['REQUEST_URI']` ?

WordPress doesn't have an explicit helper for `$_SERVER['REQUEST_URI']`, so the superglobal is fine.